ivy/functional/frontends/torch/linalg.py

```diff
@@ -229,6 +229,9 @@ def norm(input, ord=None, dim=None, keepdim=False, *, dtype=None, out=None):
         raise RuntimeError(
             f"linalg.norm: If dim is specified, it must be of length 1 or 2. Got {dim}"
         )
+    # Ensure the result is real for 'fro' and 'nuc' norms
+    if ord in ['fro', 'nuc']:
+        ret = ivy.abs(ret)
```

Comment:
```suggestion
    if dtype == "complex64":
        ret = ivy.astype(ret, "float32")
    elif dtype == "complex128":
        ret = ivy.astype(ret, "float64")
```
Looks like the only problem with the tests is the return dtype, which we can solve with this.